pkg/storage/bucket/azure/bucket_client.go

```diff
@@ -19,7 +19,10 @@ func NewBucketClient(cfg Config, name string, logger log.Logger) (objstore.Bucke
 	bucketConfig.StorageAccountName = cfg.StorageAccountName
 	bucketConfig.StorageAccountKey = cfg.StorageAccountKey.String()
 	bucketConfig.ContainerName = cfg.ContainerName
-	bucketConfig.Endpoint = cfg.Endpoint
+	if cfg.Endpoint != "" {
+		// azure.DefaultConfig has the default Endpoint
+		bucketConfig.Endpoint = cfg.Endpoint
+	}
```

Comment:
Nit, but if we did this after setting everything else, it would be cleaner than having an `if` condition in between the settings.

Reply:
Good point @colega.

Reply:
Done.